app/aqua.hoon

```diff
@@ -0,0 +1,546 @@
+::  An aquarium of virtual ships.  Put in some fish and watch them!
+::
+::  usage:
+::  |start %aqua
+::  /-  aquarium
+::  :aqua &pill .^(pill:aquarium %cx %/urbit/pill)
+::    OR
+::  :aqua &pill +solid
+::
+::  Then try stuff:
+::  :aqua [%init ~[~bud ~dev]]
+::  :aqua [%dojo ~[~bud ~dev] "[our eny (add 3 5)]"]
+::  :aqua [%dojo ~[~bud] "|hi ~dev"]
+::  :aqua [%wish ~[~bud ~dev] '(add 2 3)']
+::  :aqua [%peek ~[~bud] /cx/~bud/home/(scot %da now)/app/curl/hoon]
+::  :aqua [%dojo ~[~bud ~dev] '|mount %']
+::  :aqua [%file ~[~bud ~dev] %/sys/vane]
+::  :aqua [%pause-events ~[~bud ~dev]]
+::
+::
+::  We get ++unix-event and ++pill from /-aquarium
+::
+/-  aquarium
+=,  aquarium
+=>  $~  |%
+    +$  move  (pair bone card)
+    +$  card
+      $%  [%diff diff-type]
+      ==
+    ::
+    ::  Outgoing subscription updates
+    ::
+    +$  diff-type
+      $%  [%aqua-effects aqua-effects]
+          [%aqua-events aqua-events]
+          [%aqua-boths aqua-boths]
+      ==
+    ::
+    +$  state
+      $:  %0
+          pil=pill
+          assembled=*
+          tym=@da
+          fleet-snaps=(map term (map ship pier))
```

Comment:
Do :fleet-snaps and :piers refer to the same ships? If so, it might be cleaner to have :fleet-snaps be just a `(map term (set ship))` so there's less risk of the two getting out of sync.

Reply:
`fleet-snaps` is a set of snapshots of the state of the ships.  It's intended to freeze a fleet so you can restore back to that point later on.  This is how pH can cache e.g. "start ~bud" so that you only have to do it once rather than once per test.